lib/app/frontend/FrontendSetup.js

```diff
@@ -0,0 +1,213 @@
+const inquirer = require('inquirer')
+const os = require('os')
+const { bold, green } = require('chalk')
+const request = require('request')
+const AppSettings = require('../AppSettings')
+const logHelper = require('./LogHelper')
+const logger = require('../../logger')
+
+/**
+ * The FrontendSetup class.
+ */
+class FrontendSetup {
+  constructor () {
+    this.settings = AppSettings.getInstance()
+  }
+
+  /**
+   * Runs the setup process.
+   * @param {boolean} [showWelcomeMessage=true] If a welcome message should be shown.
+   * @return {Promise}
+   */
+  async run (showWelcomeMessage = true) {
+    if (showWelcomeMessage) {
+      logHelper.logSetupLogo()
+    }
+
+    await inquirer.prompt(this.getQuestions())
+      .then(this.registerSettings.bind(this))
+      .then(this.save.bind(this))
+      .catch(error => {
+        throw error
+      })
+  }
+
+  getQuestions () {
+    const defaultConfig = this.getDefaultConfig()
+
+    return [
+      {
+        type: 'input',
+        name: 'ip',
+        message: 'Which IP address should the app connect to?',
+        default: defaultConfig.ip
+      },
+      {
+        type: 'input',
+        name: 'port',
+        message: 'On which port should the app run?',
+        default: defaultConfig.port
+      },
+      {
+        type: 'input',
+        name: 'apiPort',
+        message: 'On which port should the Rapid API run?',
+        default: defaultConfig.apiPort
+      },
+      {
+        type: 'input',
+        name: 'hmrPort',
+        message: 'On which port should the HMR (Hot Module Replacement) run?',
+        default: defaultConfig.hmrPort
+      },
+      {
+        type: 'input',
+        name: 'remotePort',
+        message: 'On which port should the remote dev server (redux) run?',
+        default: defaultConfig.remotePort
+      },
+      {
+        type: 'input',
+        name: 'sourceMapsType',
+        message: 'Please specify your development sourcemap type:',
+        default: defaultConfig.sourceMapsType
+      },
+      {
+        type: 'input',
+        name: 'accessToken',
+        message: 'For gathering the startpage, please provide us your RAPID access token.',
+        default: defaultConfig.accessToken
+      },
+      {
+        type: 'confirm',
+        name: 'confirmed',
+        message: 'Are these settings correct?',
+        default: true
+      }
+    ]
+  }
+
+  /**
+   * Creates the request URL.
+   * @param {string} ip The IP to create the startPageUrl
+   * @param {Number} port The port to create the startPageUrl
+   * @param {string} accessToken The Jenkins access token.
+   * @param {string} appId The app id.
+   * @return {string}
+   */
+  createRequestUrl (ip, port, accessToken, appId) {
+    const startPageUrl = `http://${ip}:${port}/`
+
+    return `https://jenkins.shopgate.services:8443/buildByToken/buildWithParameters?job=SGXS_frontend_start_page_url&token=${accessToken}&START_PAGE_URL=${encodeURIComponent(startPageUrl)}&SHOP_NUMBER=${appId}`
```

Comment:
can you split this into multiple lines to make this better readable please?
e.g. encodeUriComponent can be done when intializing startPageUrl to make this return better readable